octodns/provider/azuredns.py

```diff
@@ -0,0 +1,423 @@
+#
+#
+#
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.mgmt.dns import DnsManagementClient
+from azure.mgmt.dns.models import ARecord, AaaaRecord, CnameRecord, MxRecord, \
+    SrvRecord, NsRecord, PtrRecord, TxtRecord, Zone
```

Comment:
Oh. I didn't catch the `eval` there. I'd suggest avoiding the eval in the first place and doing a `record_type` to `class` map similar to `Record.new`, [e.g.](https://github.com/github/octodns/blob/master/octodns/record.py#L80-L107). That will be explicit and traceable and will also let you get rid of the `class_name` format since you can map directly from `_type` to the supporting class.

At the [class level](https://github.com/github/octodns/pull/84/files#diff-895e411cb9ddf99d0770162b097d381dR43).

```python
TYPE_MAP = {
    'A': ARecord,
...
}
```

And then rather than the format & eval you can do

```python
self.params = self.params(record.data, key_name, self.TYPE_MAP[self.record_type])
```